src/Nancy/Extensions/TypeExtensions.cs

```diff
@@ -178,5 +203,46 @@ public static TypeCode GetTypeCode(this Type type)
             else
                 return TypeCode.Object;
         }
+
+        private static object CreateInstanceInternal(Type type, bool nonPublic = false)
+        {
+#if !DOTNET5_4
+            return Activator.CreateInstance(type, nonPublic);
+        }
+#else
+            var constructor = type.GetDefaultConstructor(nonPublic);
+
+            if (constructor == null)
+            {
+                throw new MissingMethodException("No parameterless constructor defined for this object.");
+            }
+
+            return constructor.Invoke(Array.Empty<object>());
+        }
+
+        private static ConstructorInfo GetDefaultConstructor(this Type type, bool nonPublic = false)
+        {
+            var typeInfo = type.GetTypeInfo();
+
+            var constructors = typeInfo.DeclaredConstructors;
+
+            foreach (var constructor in constructors)
+            {
+                var parameters = constructor.GetParameters();
+
+                if (parameters.Length > 0)

```

Comment:
@Yantrio a `.Lenght` should be `O(1)` since it is not a calculated property, where as `.Count()` would be slower